### PR TITLE
Feature/resource setup

### PR DESCRIPTION
**Pull request by MalasataXD**
## Feature `Resource`
This includes the full setup for Resource:
- Methods have now been added.
- `ResourceValidator` have now been added.
- Tests are now also included to ensure that validation rules are keept.

### CODE REVIEW

**Comment by MalasataXD** [comment]
Test comment for WebHook 👆

**Comment by MalasataXD** [comment]
Test comment 2 for WebHook 👍

**Comment by MalasataXD** [comment]
Test comment 3 for WebHook 🍝

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,6 @@ riderModule.iml
 *.db-wal
 **/obj/
 The-Codex-Project-BackEndV2.sln.DotSettings.user
+
+# .gitignore should also include the migration files
+Migrations/
```
Comment (BjarkeBP) [inline .gitignore]: Remove

**Review by BjarkeBP — APPROVED** [review]
LGTM :D

**Review by Grey-rgb — APPROVED** [review]
LGTM

**Pull request merged**